apps/judicial-system/backend/src/app/modules/case/case.service.ts

```diff
@@ -1019,7 +1038,10 @@ export class CaseService {
     }
 
     if (updatedCase.courtCaseNumber) {
-      if (updatedCase.courtCaseNumber !== theCase.courtCaseNumber) {
+      if (
+        updatedCase.courtCaseNumber !== theCase.courtCaseNumber &&
+        updatedCase.courtCaseNumber !== ''
```

Comment:
Please consider removing this line as this should be unnecessary. The empty string is falsy and we already checked that updatedCase.courtCaseNumber is falsy above.